code/lib/react-dom-shim/src/preset.ts

```diff
@@ -8,15 +8,15 @@ export const webpackFinal = async (config: any, options: Options) => {
 
   const isReact18 = version.startsWith('18') || version.startsWith('0.0.0');
   const useReact17 = legacyRootApi ?? !isReact18;
-  if (useReact17) return config;
+  if (!useReact17) return config;
 
   return {
     ...config,
     resolve: {
       ...config.resolve,
       alias: {
         ...config.resolve?.alias,
-        '@storybook/react-dom-shim': '@storybook/react-dom-shim/dist/react-18',
+        '@storybook/react-dom-shim': '@storybook/react-dom-shim/dist/react-16',
       },
     },
   };
```

Comment:
something seems confusing to me in this file, `useReact17` but then shim with React 16? Can we clear that up somehow?

Reply:
Yes, great question, I do believe the "react17" part is very confusing, but that already existed.
I can explain what's going on:

This is about docs/preview.

The default is now react18.
You can see that by looking at the package.json of this `@storybook/react-dom-shim`.

So we do NOT have to alias when the user is using react18.
We do need to alias if the user is using react16 or 17

There's also a feature flag to force the legacyRootApi (which does no longer exist in react18, so that only applies to react17, because react16 can only do the legacyRoot thing.